edm4eic.yaml

```diff
@@ -544,3 +544,15 @@ datatypes:
      - edm4eic::RawTrackerHit rawHit        // reference to the digitized hit
     OneToManyRelations:
      - edm4hep::SimTrackerHit simHits       // reference to the simulated hits
+
+
+  edm4eic::MCParticleRecoHitAssociation:
```

Comment:
If the association is known at the time when `edm4eic::TrackerHit` is originally created, just add the `mcParticles` field to it, don't make another collection.

Reply:
This uses raw hit, we want tracker hits. I don't think it's a good ideas to add the mcparticle here and then create another layer of link to go from tracker hit to raw hits. Let's just have a new association to provide the direct tracker hits to mcparticles link which is exactly what will be used in analysis.

Reply:
I don't understand your explanation. Line 555 reads TrackerHit. Are you responding to @wdconinc's comment?

Reply:
Ahh sorry I thought you suggested to use the existing association structure on line 546 and add mcparticles there. If you mean adding mcparticles directly to trackerhit instead of creating a separate association, then we are having our reconstructing algorithm directly linked to mc information, which is not recommended according to Wouter

Reply:
That's a good point. Keeping MC associations as separate collections allows to be very explicit about whether those valuables are available. However, optional factory inputs/outputs are not implemented and there is no clear understanding of how that should work even, so this policy may not be very sound in the long run.

I will mark this as resolved.